src/Build/BackEnd/Shared/BuildResult.cs

```diff
@@ -562,9 +562,9 @@ internal static BuildResult FactoryForDeserialization(ITranslator translator)
         /// </summary>
         internal void CacheIfPossible()
         {
-            foreach (string target in _resultsByTarget.Keys)
+            foreach (KeyValuePair<string, TargetResult> targetResultPair in _resultsByTarget)
             {
-                _resultsByTarget[target].CacheItems(ConfigurationId, target);
```

Comment:
Nice! I wish there was a code analyzer flagging this pattern.

Reply:
❤️

(This pattern was my initial impetus for the whole PR; the rest was when I got distracted)